src/ILCompiler.Compiler/src/Compiler/Compilation.cs

```diff
@@ -102,7 +102,7 @@ public ObjectNode GetFieldRvaData(FieldDesc field)
             {
                 var pInvokeFixup = (PInvokeLazyFixupField)field;
                 PInvokeMetadata metadata = pInvokeFixup.PInvokeMetadata;
-                return NodeFactory.PInvokeMethodFixup(metadata.Module, metadata.Name);
+                return NodeFactory.PInvokeMethodFixup(metadata.Module, metadata.Name, metadata.Flags.ExactSpelling, metadata.Flags.CharSet);
```

Comment:
nit: could you pass PInvokeFlags directly instead of its two Properties? since it is likely We will use its others Properties.